### ISSUE TITLE
Damaging Gyth while Flame Breath prevent Rend to drop off

### ISSUE THREAD

**Issue by Shineslippers** [azerothcore/azerothcore-wotlk#9062]
### What client do you play on?

enUS

### Faction

- [X] Alliance
- [X] Horde

### Content Phase:

- [ ] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [X] 50-59

### Current Behaviour

Damaging [Gyth](https://wowgaming.altervista.org/aowow/?npc=10339) while Flame Breath is casting by him near drop off script will call Rend to disappear.

https://user-images.githubusercontent.com/85256266/140809229-a4f11180-90b4-436c-9695-ec0f74cadf91.mp4

### Expected Blizzlike Behaviour

Rend should drop off or it will leave doors closed because event not finished

### Source

None.

### Steps to reproduce the problem

.go c 137967
start Gyth and Rend event

### AC rev. hash/commit

https://github.com/chromiecraft/azerothcore-wotlk/commit/420ca2e2b57216b90f93e2deb870946b420a72df

### Operating system

Ubuntu 20.04

### Modules

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-chat-transmitter](https://github.com/azerothcore/mod-chat-transmitter)
- [mod-chromie-xp](https://github.com/azerothcore/mod-chromie-xp)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-ip-tracker](https://github.com/azerothcore/mod-ip-tracker)
- [mod-low-level-arena](https://github.com/azerothcore/mod-low-level-arena)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-pvpstats-announcer](https://github.com/azerothcore/mod-pvpstats-announcer)
- [mod-queue-list-cache](https://github.com/azerothcore/mod-queue-list-cache)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-carbon-copy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-custom-worldboss](https://github.com/55Honey/Acore_CustomWorldboss)
- [lua-level-up-reward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-recruit-a-friend](https://github.com/55Honey/Acore_RecruitAFriend)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)
- [lua-temp-announcements](https://github.com/55Honey/Acore_TempAnnouncements)
- [lua-zonecheck](https://github.com/55Honey/acore_Zonecheck)
- [lua-zone-debuff](https://github.com/55Honey/Acore_ZoneDebuff)

### Customizations

None

### Server

ChromieCraft

**Comment by Shineslippers** [comment]
OOops it could be intended behavior. Atleast this saying [Wowwiki](https://wowwiki-archive.fandom.com/wiki/Dal%27rend_Blackhand?oldid=2396248#Strategies):
>At higher levels, it is of paramount importance that Rend be brought down slowly. If he is brought down too fast, he will not get a chance to jump off of his mount (which he does when he has about 25% of his health remaining, if you DPS him down first) and, though Rend will be lootable, the Stadium door will remain closed, and necessitate that the instance be reset. This also applies if you choose to DPS down Gyth first; Rend drops off him when Gyth has around 25% health remaining, and any attack that kills Gyth before Rend dismounts will have the same effect, except that Rend will despawn instead of leaving a corpse to be looted (Gyth can be looted as normal, however).

**Comment by Nyeriah** [comment]
No, Rend is summoned by a spell and it needs to have triggered cast flag so the spell cast doesn't fail because the dragon is already casting something.

The "bug" you described above is not present @ AC, he'll always spawn Gyth before dying, I fixed that last week. If he's supposed to die before spawning Rend I can "break" it again if necessary.

**Comment by Shineslippers** [comment]
For my opinion better to fix any possible drop off bugs. Because if for some reason Rend will not be available to kill it will make everyone sad, even if it's blizzlike.
